src/pymatgen/analysis/adsorption.py

```diff
@@ -664,6 +662,10 @@ def plot_slab(
         decay (float): how the alpha-value decays along the z-axis
         inverse (bool): invert z axis to plot opposite surface
     """
+    # Expensive import (PR4128)
+    from matplotlib import patches
```

Comment:
`matplotlib` is almost certainly not a core module for pymatgen.core, and it incurred significant import overhead to `core.interface`:
<img width="1475" alt="image" src="https://github.com/user-attachments/assets/c1d02fab-e5bd-44c1-b5c5-d49af88eb018">

After: 
<img width="1475" alt="image" src="https://github.com/user-attachments/assets/86e6d0d7-3f7b-4470-858a-7b00a02cebdd">